lib/components/admin-command.js

```diff
@@ -0,0 +1,157 @@
+"use strict";
+
+var minimist = require("minimist");
+
+/**
+ * Construct a new AdminCommand helper instance.
+ * @constructor
+ * @param {string} opts.desc A short description to display in the 'help'
+ * command.
+ * @param {AdminCommand~opts=} opts.opts An optional object giving commandline
+ * options recognised by the command.
+ * @param {AdminCommand~func} opts.func The handler function that implements
+ * the command.
+ */
+function AdminCommand(opts) {
+    this.desc = opts.desc;
+    this._func = opts.func;
+
+    if (opts.opts) {
+        this.optspec = {};
```

Comment:
Your intention here is to make `optspec` and `optaliases` be maps. Sadly, this is Javascript and `Objects` are poor map replacements because of this:
```js
> s = {}
{}
> s["hasOwnProperty"]
[Function: hasOwnProperty]
> "hasOwnProperty" in s
true
```

The short-hand ` = {}` will use the default `Object.prototype` which has a bunch of keys already inserted into your Map! You can get around this by *explicitly* creating an object without a prototype:

```js
> s = Object.create(null);
{}
> s["hasOwnProperty"]
undefined
> "hasOwnProperty" in s
false
```

This is particularly important when you don't control what the key names are called (as in this case).